subsys/bluetooth/services/hrs_client.c

```diff
@@ -0,0 +1,414 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <bluetooth/services/hrs_client.h>
+#include <sys/util.h>
+#include <sys/byteorder.h>
+
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(hrs_client, CONFIG_BT_HRS_CLIENT_LOG_LEVEL);
+
+#define HRS_MEASUREMENT_NOTIFY_ENABLED BIT(0)
+#define HRS_MEASUREMENT_READ_IN_PROGRES BIT(1)
+#define HRS_SENSOR_LOCATION_READ_IN_PROGRES BIT(2)
+#define HRS_CONTROL_POINT_WRITE_PENDING BIT(3)
+
+#define HRS_MEASUREMENT_FLAGS_VALUE_FORMAT      BIT(0)
+#define HRS_MEASUREMENT_FLAGS_CONTACT_DETECTED  BIT(1)
+#define HRS_MEASUREMENT_FLAGS_CONTACT_SUPPORTED BIT(2)
+#define HRS_MEASUREMENT_FLAGS_ENERGY_EXPENDED   BIT(3)
+#define HRS_MEASUREMENT_FLAGS_RR_INTERVALS      BIT(4)
+
+static bool data_length_check(uint16_t current, uint16_t min_expected)
+{
+	return (current >= min_expected);
+}
+
+static void hrs_reinit(struct bt_hrs_client *hrs_c)
+{
+	hrs_c->measurement_char.handle = 0;
+	hrs_c->measurement_char.ccc_handle = 0;
+	hrs_c->measurement_char.notify_cb = NULL;
+
+	hrs_c->sensor_location_char.handle = 0;
+	hrs_c->sensor_location_char.read_cb = NULL;
+
+	hrs_c->cp_char.handle = 0;
+
+	hrs_c->conn = NULL;
+	hrs_c->state = ATOMIC_INIT(0);
```

Comment:
Can you do this in one step using memset?

Reply:
There is not need to clean all instance data, that is way I didn't use memset